tsinfer/formats.py

```diff
@@ -765,11 +766,30 @@ class Site:
     # TODO document properly.
     id = attr.ib()
     position = attr.ib()
-    ancestral_state = attr.ib()
+    ancestral_allele = attr.ib()  # here -1 (tskit.MISSING_DATA) means none defined
     metadata = attr.ib()
     time = attr.ib()
     alleles = attr.ib()
 
+    @property
+    def ancestral_state(self):
+        if self.ancestral_allele == MISSING_DATA:
+            return None
+        return self.alleles[self.ancestral_allele]
+
+    @property
```

Comment:
I think this is a function rather than a property. ``site.reorder_alleles()`` or something?